Dynamic protocols switching (#113)
nrfconnect/ncs-zigbee: docs/architectures.rst

===========================================

In this design, the Zigbee (ZBOSS) and Matter over Thread (OpenThread) stacks run on the same nRF SoC and share the single 802.15.4 radio through a coexistence layer, while a `SoftDevice Controller`_ instance handles Bluetooth LE (CHIPoBLE) commissioning.

> Radio ownership is time-separated and persisted across reboots: the device boots on the configured default protocol (Zigbee by default), advertises for Matter commissioning over Bluetooth LE in parallel when Zigbee is active, and hands the 802.15.4 radio over to OpenThread once Matter commissioning completes or the user requests a protocol switch.

Reviewer: Simplified
```suggestion
Radio ownership is time-separated and persists across reboots. 
The device boots on the configured default protocol (Zigbee by default).
While Zigbee is active, Matter advertises for commissioning over Bluetooth LE in parallel.
The 802.15.4 radio is handed over to OpenThread once Matter commissioning completes or you request a protocol switch.
```
